src/core/createFrameResponse.ts

```diff
@@ -0,0 +1,58 @@
+import { NextResponse } from 'next/server';
+import { Button, FrameMetadata } from './sharedTypes';
+
+/**
+ * Creates a response with a new valid frame to be rendered.
+ *
+ * @param buttons: The buttons to use for the frame.
+ * @param image: The image to use for the frame.
+ * @param post_url: The URL to post the frame to.
+ * @param refresh_period: The refresh period for the image used.
+ * @returns An HTML response containing metadata for the frame.
+ */
+function createFrameResponse({
```

Comment:
There is something about this name that I am not fully sold.

Part of me thinks `getFrameReponse` is better, as you are describing the end goal vs the action.

Also I am a bit overthinknig if something like `getFrameNextResponse` is more correct. As the type returned is `NextResponse`.

@taycaldwell  @cnasc @robpolak thoughts?

Reply:
Removed the dependency on Next.js by having it return the HTML string instead of wrapping it in a NextResponse and returning that.

Because of this and the suggestions above, I updated the name to `getFrameHtml`.